### ISSUE TITLE
no-useless-escape is missing \B regex escape

### ISSUE THREAD

**Issue by platinumazure** [eslint/eslint#5750]
**What version of ESLint are you using?**

2.5.3

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```json
{
    "rules": {
        "no-useless-escape": "error"
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
var regex = /.\B./;
```

**What did you expect to happen?**

No error-- `\B` is a valid regex escape, meaning the inverse of `\b` (that is, `\B` is a zero-width assertion that the characters on either side are both in `\w` or both in `\W`).

**What actually happened? Please include the actual, raw output from ESLint.**

The character was noted as a supposedly "invalid" escape.

--------

I went through MDN and found no other missing escapes, so it's just `\B`. This is an [assertion production](https://tc39.github.io/ecma262/#prod-Assertion) in the ECMAScript standard.

**Comment by mysticatea** [comment]
![image](https://cloud.githubusercontent.com/assets/1937871/14199631/1a317f20-f821-11e5-93c8-c0275b23fa55.png)
![image](https://cloud.githubusercontent.com/assets/1937871/14199650/35765670-f821-11e5-868c-2af522c7b8d4.png)
